### PR TITLE
Refactoring Study model for maintainability (SCP-2771)

### PR DESCRIPTION
**Pull request by bistline**
The `study.rb` model is by far the largest class at 3400+ lines, and contains many misplaced or deprecated methods.  This change removes deprecated methods/tests, and migrates remaining misplaced code into more appropriate classes.  The result has removed 1500+ lines from `study.rb`, a reduction in size of over 44%.

This PR satisfies SCP-2771

### CODE REVIEW

**Review by eweitz — APPROVED** [review]
Looks good!  It seems the vast majority of the removed code is now handled by Ingest Pipeline.

**Review by devonbush — APPROVED** [review]
Nice!  I assume you cut-and-paste correctly, so I focused my effort on reading the error/log messages to make sure that no references to 'study' were missing (which is what burned me when I first moved some stuff to parse_utils a few months ago).

File: app/lib/file_parse_service.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
+extend ErrorTracker
```
[the remark this answers is not in the document]
Comment (bistline) [reply app/lib/file_parse_service.rb]: To be honest, I don't know why I put that in there either... I think I got my methods mixed up re: modules.  9b63d6f

**Review by knapii-developments — COMMENTED** [review]
Thanks for walking me through this. This refactoring will make methods easier to find while debugging.

**Pull request merged**